### DISCUSSION TITLE
getting error: each.value is object with 2 attributes

### DISCUSSION THREAD

**Discussion by hptltech** [learndevopsonline/learndevopsonline#1993] [Q&A] [answered]
I am deploying roboshop-terrafom using github-runner, deploy roboshop in kubernetes cluster.
While executing code getting this error:
╷
│ Error: Invalid index
│ 
│   on main.tf line 39, in module "web_instances":
│   39:   volume_size    = each.value["volume_size"]
│     ├────────────────
│     │ each.value is object with 2 attributes
│ 
│ The given key does not identify an element in this collection value.

**here is terraform code:**

module "web_instances" {
  depends_on     = [module.app_instances]
  for_each       = var.web_instances
  source         = "./modules/ec2"
  env            = var.env
  app_port       = each.value["app_port"]
  component_name = each.key
  instance_type  = each.value["instance_type"]
  domain_name    = var.domain_name
  zone_id        = var.zone_id
  vault_token    = var.vault_token
  volume_size    = each.value["volume_size"]
}

Appreciate your help

**Comment by chaithanya1812** [comment]
![image](https://github.com/user-attachments/assets/be6fcd02-17f7-4396-a409-ea9e4f04c7b0)

**Reply by hptltech** [reply, marked as answer]
web_instances = {
  frontend = {
    app_port      = 80
    instance_type = "t3.small"
    volume_size   = "20"
  }
}

this was already updated, was still getting same error

**Reply by hptltech** [reply]
Thanks Chaitnya! obviously, the workflow needed to run again after commit (can't do rerun failed job)
